### ISSUE TITLE
Different results for same primitives and different max_depth in featuretools.dfs

### ISSUE THREAD

**Issue by MarekHauzr** [Featuretools/featuretools#317]
### Different results for same primitives and different max_depth in featuretools.dfs

When using different max_depth I get different results for the same combinations of primitives. Which I think should be the same.

Setup:
```
from featuretools.primitives import Percentile, CumMean, CumSum
import featuretools # version 0.4.0, same results for 0.3.1
import numpy as np # version 1.15.1
import pandas as pd # version 0.23.4

sine_wave = pd.DataFrame({
    'sine': np.sin(np.arange(10)),
    'timestamp': pd.DatetimeIndex(start = '1/1/2018', freq = '1s', periods = 10),
    'row_id': np.arange(10),
    'category': 0
})

es = featuretools.EntitySet(id = 'sine_wave_timeseries')
es.entity_from_dataframe(
    entity_id = 'sine_wave',
    dataframe = sine_wave,
    index = 'row_id',
    time_index = 'timestamp',
    variable_types = {
        'sine': featuretools.variable_types.Numeric,
        'timestamp': featuretools.variable_types.DatetimeTimeIndex,
        'row_id': featuretools.variable_types.Categorical,
        'category': featuretools.variable_types.Categorical,
    }
)
es.normalize_entity(
    base_entity_id = 'sine_wave',
    new_entity_id = 'observations',
    index = 'category',
    make_time_index = True
)
es.add_last_time_indexes(['observations', 'sine_wave'])

cutoffs = sine_wave[['row_id', 'timestamp']]
```
Using `max_depth = 3` and `max_depth = 2`
```
feature_matrix_3, _ = featuretools.dfs(
    entityset = es,
    target_entity = 'sine_wave',
    cutoff_time = cutoffs,
    cutoff_time_in_index = True,
    trans_primitives = [CumMean, CumSum, Percentile],
    agg_primitives = [],
    training_window = '3 seconds',
    max_depth = 3
)
feature_matrix_2, _ = featuretools.dfs(
    entityset = es,
    target_entity = 'sine_wave',
    cutoff_time = cutoffs,
    cutoff_time_in_index = True,
    trans_primitives = [CumMean, CumSum, Percentile],
    agg_primitives = [],
    training_window = '3 seconds',
    max_depth = 2
)
```
and the output when looking at the `'CUM_SUM(CUM_MEAN(sine by category) by category)'` is different for the different `max_depth`
```
feature_matrix_3[[ 
     'CUM_SUM(CUM_MEAN(sine by category) by category)' 
]]
#                            CUM_SUM(CUM_MEAN(sine by category) by category)
#row_id time                                                                
#0      2018-01-01 00:00:00                                         0.000000
#1      2018-01-01 00:00:01                                         0.420735
#2      2018-01-01 00:00:02                                         1.004325
#3      2018-01-01 00:00:03                                         1.477297
#4      2018-01-01 00:00:04                                         2.631256
#5      2018-01-01 00:00:05                                         1.366050
#6      2018-01-01 00:00:06                                        -1.155096
#7      2018-01-01 00:00:07                                        -2.614252
#8      2018-01-01 00:00:08                                        -1.669877
#9      2018-01-01 00:00:09                                         0.809775

feature_matrix_2[[ 
     'CUM_SUM(CUM_MEAN(sine by category) by category)' 
]]
#                            CUM_SUM(CUM_MEAN(sine by category) by category)
#row_id time                                                                
#0      2018-01-01 00:00:00                                         0.000000
#1      2018-01-01 00:00:01                                         0.420735
#2      2018-01-01 00:00:02                                         0.583589
#3      2018-01-01 00:00:03                                         0.472972
#4      2018-01-01 00:00:04                                         0.283771
#5      2018-01-01 00:00:05                                        -0.166327
#6      2018-01-01 00:00:06                                        -0.463506
#7      2018-01-01 00:00:07                                        -0.334539
#8      2018-01-01 00:00:08                                         0.102001
#9      2018-01-01 00:00:09                                         0.444762

```

**Comment by kmax12** [comment]
The issue appears to be calculating this feature alongside other features that use the `Percentile` primitive. If you remove the `Percentile` primitive from your example, the discrepancy goes away. 

Thanks for reporting. We are looking into it now and will keep you posted.

**Comment by kmax12** [comment]
@MarekHauzr we just put up a PR to fix this in #320. Can you test it out and confirm if it works for you? 

You can install that specific branch with this command

```
pip install -e git://github.com/featuretools/featuretools.git@fix-cum-primitives#egg=featuretools
```

**Comment by MarekHauzr** [comment]
@kmax12 Thanks, it works now.
